### PR TITLE
Tags

### PR DESCRIPTION
**Pull request by shakhzodniyozov**

### CODE REVIEW

File: src/Infrastructure/Data/Configurations/TagConfiguration.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    }
+
+    protected override void RegisterClassMap(BsonClassMap<Tag> classMap)
+    {
+        base.RegisterClassMap(classMap);
+        classMap.MapMember(x => x.Name).SetElementName("name");
+        classMap.MapMember(x => x.Color).SetElementName("color");
+    }
+
+    protected override void Configure() { }
```
Comment (SidiqovAbbos) [inline src/Infrastructure/Data/Configurations/TagConfiguration.cs]: It should be optional to implement

File: src/Web/Server/Controllers/TagController.cs [hunk 2]
```diff
@@ -0,0 +1,40 @@
+using Microsoft.AspNetCore.Mvc;
+
+using MRA.AssetsManagement.Application.Features.Tags.Commands;
+using MRA.AssetsManagement.Application.Features.Tags.Queries;
+using MRA.AssetsManagement.Domain.Entities;
+
+namespace MRA.AssetsManagement.Web.Server.Controllers;
+
+public class TagController : ApiControllerBase
```
Comment (SidiqovAbbos) [inline src/Web/Server/Controllers/TagController.cs]: We had discussion about putting ProducesResponseType to the actions. 
What is the status of this?

image

File: src/Infrastructure/Data/Configurations/TagConfiguration.cs (hunk 1 above)
Comment (shakhzodniyozov) [reply src/Infrastructure/Data/Configurations/TagConfiguration.cs]: Is it a solution if I change signature of Configure method from abstract to virtual? Class that derives from BaseConfiguration overrides the Configure method when it is necessary.

File: src/Web/Server/Controllers/TagController.cs (hunk 2 above)
Comment (shakhzodniyozov) [reply src/Web/Server/Controllers/TagController.cs]: In progress. I'm learning it.

**Review by SidiqovAbbos — COMMENTED** [review]
Tags feature!
image


Not found should be 404 without throwing an exception

image

**Comment by shakhzodniyozov** [comment]
Ака CR га тайер

File: src/Application/Common/Exceptions/NotFoundException.cs [hunk 3]
```diff
@@ -0,0 +1,11 @@
+namespace MRA.AssetsManagement.Application.Common.Exceptions;
+
+public class NotFoundException : Exception
```
Comment (SidiqovAbbos) [inline src/Application/Common/Exceptions/NotFoundException.cs]: `NotFoundEntityException`

File: src/Application/Common/Exceptions/NotFoundException.cs [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+{
+    public NotFoundException() { }
+    
+    public NotFoundException(string message)
```
Comment (SidiqovAbbos) [inline src/Application/Common/Exceptions/NotFoundException.cs]: public NotFoundException(T entity, string id) : base("${typeof(T).Name] with provided ${id} was not found")

File: src/Application/Features/AssetTypes/Commands/RestoreAssetTypeCommand.cs [hunk 5]
```diff
@@ -16,12 +16,12 @@ public RestoreAssetTypeCommandHandler(IApplicationDbContext context)
 
     public async Task Handle(RestoreAssetTypeCommand request, CancellationToken cancellationToken)
     {
-        var assetType = await _context.AssetTypes.GetAsync(request.Id);
+        var assetType = await _context.AssetTypes.GetAsync(request.Id, cancellationToken);
 
         if (assetType is null)
             throw new Exception("AssetType with provided Id was not found.");
```
Comment (SidiqovAbbos) [inline src/Application/Features/AssetTypes/Commands/RestoreAssetTypeCommand.cs]: NotFoundEntityException

File: src/Application/Features/AssetTypes/Commands/UpdateAssetTypeCommand.cs [hunk 6]
```diff
@@ -25,13 +25,13 @@ public UpdateAssetTypeCommandHandler(IApplicationDbContext context, IMapper mapp
     
     public async Task Handle(UpdateAssetTypeCommand request, CancellationToken cancellationToken)
     {
-        var assetType = await _context.AssetTypes.GetAsync(request.Id);
+        var assetType = await _context.AssetTypes.GetAsync(request.Id, cancellationToken);
         
         if (assetType is null)
             throw new Exception("AssetType with provided Id was not found.");
```
Comment (SidiqovAbbos) [inline src/Application/Features/AssetTypes/Commands/UpdateAssetTypeCommand.cs]: NotFound exception

File: src/Web/Server/Controllers/TagController.cs [hunk 7]
```diff
@@ -0,0 +1,49 @@
+using Microsoft.AspNetCore.Mvc;
+using MRA.AssetsManagement.Application.Features.Tags.Commands;
+using MRA.AssetsManagement.Application.Features.Tags.Queries;
+using MRA.AssetsManagement.Domain.Entities;
+
+namespace MRA.AssetsManagement.Web.Server.Controllers;
+
+public class TagController : ApiControllerBase
```
Comment (SidiqovAbbos) [inline src/Web/Server/Controllers/TagController.cs]: TagsController

**Review by SidiqovAbbos — APPROVED** [review]

**Pull request merged**